internal/service/glue/job.go

```diff
@@ -207,6 +218,16 @@ func resourceJobCreate(ctx context.Context, d *schema.ResourceData, meta interfa
 		Tags:    getTagsIn(ctx),
 	}
 
+	// Add job_mode to the input (assuming it might map to DefaultArguments or a future API field)
+	if v, ok := d.GetOk("job_mode"); ok {
+		jobMode := v.(string)
+		if input.DefaultArguments == nil {
+			input.DefaultArguments = make(map[string]string)
+		}
+		// For now, store job_mode in DefaultArguments as a workaround
+		input.DefaultArguments["--job-mode"] = jobMode
+	}
+
```

Comment:
JobMode is not an argument included in DefaultArguments. It's a separate API parameter. See https://docs.aws.amazon.com/glue/latest/webapi/API_CreateJob.html